src/Cli/dotnet/CommonOptions.cs

```diff
@@ -296,15 +294,17 @@ internal static string ResolveRidShorthandOptionsToRuntimeIdentifier(string os,
     public static string GetCurrentRuntimeId()
     {
         // Get the dotnet directory, while ignoring custom msbuild resolvers
-        string dotnetRootPath = NativeWrapper.EnvironmentProvider.GetDotnetExeDirectory(key =>
+        string? dotnetRootPath = NativeWrapper.EnvironmentProvider.GetDotnetExeDirectory(key =>
             key.Equals("DOTNET_MSBUILD_SDK_RESOLVER_CLI_DIR", StringComparison.InvariantCultureIgnoreCase)
                 ? null
                 : Environment.GetEnvironmentVariable(key));
         var ridFileName = "NETCoreSdkRuntimeIdentifierChain.txt";
+        var sdkPath = dotnetRootPath is not null ? Path.Combine(dotnetRootPath, "sdk") : "sdk";
```

Comment:
This is a case where the NRT analysis seems to have spotted a legitimate bug in the code. In the case that `dotnetRootpath` is `null` then the `Path.Combine` calls below will throw. I changed this to just use `"sdk"` as a relative path but possible that a `throws` here is a better idea.